examples/case-management.v1.test.js

```diff
@@ -0,0 +1,422 @@
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation(output => {
+      done(output);
+    });
+
+    expect(caseNumber).not.toBeNull();
+
+    // begin-removeWatchlist
+
+    const watchlistUsers = [
+      {realm: 'IBMid', user_id: 'abc@ibm.com'}
+    ]
+
+    const params = {
+      caseNumber: caseNumber,
+      watchlist: watchlistUsers,
+    };
+
+    caseManagementService.removeWatchlist(params)
+      .then(res => {
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch(err => {
+        console.warn(err)
+      });
+
+    // end-removeWatchlist
+  });
+  test('addResource request example', done => {
+
+    consoleLogMock.mockImplementation(output => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation(output => {
+      done(output);
+    });
+
+    expect(caseNumber).not.toBeNull();
+    expect(resourceCrn).not.toBeNull();
+
+    // begin-addResource
+
+    const params = {
+      caseNumber: caseNumber,
+      crn: resourceCrn,
+      note: 'This resource is the service that is having the problem.',
+    };
+
+    caseManagementService.addResource(params)
+      .then(res => {
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch(err => {
+        console.warn(err)
+      });
+
+    // end-addResource
+  });
+  test('uploadFile request example', done => {
+
+    consoleLogMock.mockImplementation(output => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation(output => {
+      done(output);
+    });
+
+    expect(caseNumber).not.toBeNull();
+
+    // begin-uploadFile
+
+    const exampleFileContent = 'This is the content of the file to upload.';
+
+    const fileWithMetadata = {
+      data: Buffer.from(exampleFileContent),
+      filename: 'example.log',
+      content_type: 'application/octet-stream',
+    };
+
+    const filesToUpload = [fileWithMetadata]
+
+    const params = {
+      caseNumber: caseNumber,
+      file: filesToUpload,
+    };
+
+    caseManagementService.uploadFile(params)
+      .then(res => {
+        attachmentId = res.result.id;
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch(err => {
+        console.warn(err)
+      });
+
+    // end-uploadFile
+  });
+  test('downloadFile request example', done => {
+
+    consoleLogMock.mockImplementation(output => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation(output => {
+      done(output);
+    });
+
+    expect(caseNumber).not.toBeNull();
+    expect(attachmentId).not.toBeNull();
+
+    // begin-downloadFile
+
+    const params = {
+      caseNumber: caseNumber,
+      fileId: attachmentId,
+    };
+
+    caseManagementService.downloadFile(params)
+      .then(res => {
+
+        console.log(`Attachment content-type: ${res.headers['content-type']}\nAttachment contents: ${res.result.read()}`);
```

Comment:
I don't know why this way of reading the response body (a stream) works for you.   I tried the exact same thing in the global catalog examples and it didn't work.   Dustin helped me fix that by using the node core's "streamToPromise()" method.   You can see that here: https://github.com/IBM/platform-services-node-sdk/blob/8315ebeab477b1cc7427906bff291eb92cd1a782/examples/global-catalog.v1.test.js#L448

I think it would be good for you to change this test block to work the same way, because reading the response body is inherently async and so we should probably do it in an async fashion.  I think you're just getting lucky that it works here :)

Reply:
Fixed in https://github.com/IBM/platform-services-node-sdk/pull/77/commits/61230f6c97e95f14d1e34db1aaaf443db2c33b3b